File: user/src/main/java/ss/user/controller/ro/AccountRO.java
package ss.user.controller.ro;


import org.springframework.security.core.GrantedAuthority;
import ss.user.atcl.AuthConverter;
import ss.user.domain.User;
import ss.user.domain.UserSummary;

import java.time.LocalDateTime;
import java.util.List;

public record AccountRO(
        Long id,
        String username,
        String nickname,
        String avatar,
        List<String> authorities,
        LocalDateTime createTime,
        LocalDateTime updateTime,
        boolean accountNonExpired,
        boolean accountNonLocked,
        boolean enabled
) {
    public static AccountRO create(User user, UserSummary userSummary) {
        return new AccountRO(
                user.getId(),
                user.getUsername(),
                userSummary.getNickname(),
                userSummary.getAvatar(),
                AuthConverter.mapBitToRole(user.getAuthorities()).stream().map(GrantedAuthority::getAuthority).toList(),
                user.getCreateTime(),
                user.getUpdateTime(),
                user.isAccountNonExpired(),
                user.isAccountNonLocked(),
                user.isEnabled()
        );
    }
}

File: quasar/src/domain/AccountRO.ts
export type AccountRO = {
  id: string; // 用户ID
  username: string; // 用户名
  nickname: string; // 昵称
  avatar: string; // 头像URL
  authorities: string[]; // 权限列表
  createTime: Date; // 创建时间
  updateTime: Date; // 更新时间
  accountNonExpired: boolean; // 是否未过期
  accountNonLocked: boolean; // 是否未被锁定
  enabled: boolean; // 是否启用
};
